### PR TITLE
Auto scope

### PR DESCRIPTION
**Pull request by Frikki**
### chore(package): add and update dependencies …

Update:
@cycle/core 3.1.x => 6.0.0-rc2
@cycle/dom 5.1.x => 8.0.0-rc2

Add:
@cycle/isolate 1.0.x
rx ^4.0.6
### refactor(many): update with Rx and @cycle/dom …

Import named exports from Rx.
Use hyperscript helpers.
Rename _list_ dataflow component to _List_.
### feat(isolate): use isolate and rewrite to dataflow components …
- Capitalize components and their respective files.
- Isolate `Item` in `itemWrapper`.
- Remove redundant `name` array.
- Remove redundant CSS selectors in `intent`.

### CODE REVIEW

**Comment by staltz** [comment]
Good job. When I was working on advanced-list-nest example, I noticed one way I can refactor the new dataflow component to not have any `name` or `id` secondary parameter. Like this https://github.com/cyclejs/cycle-examples/blob/auto-scope/advanced-list-nest/src/app.js#L56-L62 . Do you think you could update this PR to have the same approach? This way we don't leak parent concern ("_do instances of my child dataflow component have ids attached to them so I can query them separately?_") into the child ("_I think I need an id for some reason_").

**Comment by Frikki** [comment]
Indeed.

File: many/src/List.js [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+  const addItem$ = Observable.merge(
+    DOM.select('.add-one-btn')
       .events('click').map(() => 1),
-    DOM.select(name.join(' ') + '.list .add-many-btn')
+    DOM.select('.add-many-btn')
       .events('click').map(() => 1000)
   );
   const removeItem$ = itemActions.destroy$
-    .map(({name}) => parseInt(name[name.length-1].replace('.list-item', '')));
+    .map(id => id);
```
Comment (staltz) [inline many/src/List.js]: Unnecessary map here.

File: many/src/List.js [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
+import isolate from '@cycle/isolate'
+import Item from './Item';
 
-function intent(DOM, itemActions, name = []) {
-  const addItem$ = Rx.Observable.merge(
-    DOM.select(name.join(' ') + '.list .add-one-btn')
+function intent(DOM, itemActions) {
+  const addItem$ = Observable.merge(
+    DOM.select('.add-one-btn')
       .events('click').map(() => 1),
```
Comment (staltz) [inline many/src/List.js]: Maybe this can be written in just one line?

File: many/src/List.js [hunk 3]
```diff
@@ -59,54 +60,60 @@ function model(actions, itemFn) {
...
+function makeItemWrapper(DOM) {
   return function itemWrapper(props, id) {
     const propsObservables = {
-      color$: Rx.Observable.just(props.color),
-      width$: Rx.Observable.just(props.width),
+      color$: Observable.just(props.color),
+      width$: Observable.just(props.width),
     };
-    return item({DOM, props: propsObservables}, name.concat(`.list-item${id}`));
+    const item = isolate(Item, `${id}`)({
```
Comment (staltz) [inline many/src/List.js]: You don't need anymore `${id}` parameter here.

Comment (Frikki) [inline many/src/List.js]: Ah, yes. I missed that. Fixing.

**Comment by staltz** [comment]
Great, thanks

**Pull request merged**